internal/cmd/local/command_kafka_start.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/phayes/freeport"
+	"github.com/spf13/cobra"
+
+	v1 "github.com/confluentinc/cli/internal/pkg/config/v1"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+func (c *command) newKafkaStartCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "start",
+		Short: "Start a single-node instance of Apache Kafka.",
+		Args:  cobra.NoArgs,
+		RunE:  c.kafkaStart,
+	}
+}
+
+func (c *command) kafkaStart(cmd *cobra.Command, args []string) error {
+	dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+	defer dockerClient.Close()
+
+	if err := checkIsDockerRunning(dockerClient); err != nil {
+		return err
+	}
+
+	// pull the image from ecr. it will be public so no creds needed?
+	// out, err := cli.ImagePull(ctx, imageName, types.ImagePullOptions{})
+	// if err != nil {
+	// 	return err
+	// }
+	// defer out.Close()
+	// io.Copy(os.Stdout, out)
+	// log.CliLogger.Tracef("Pull confluent-local image success")
+
+	err = c.prepareValidPorts()
+	if err != nil {
+		return err
+	}
+
+	if c.Config.LocalPorts == nil {
+		return errors.NewErrorWithSuggestions(errors.FailedToReadPortsErrorMsg, errors.FailedToReadPortsSuggestions)
+	}
+	ports := c.Config.LocalPorts
+	platform := &specsv1.Platform{OS: "linux", Architecture: runtime.GOARCH}
+	natRestPort := nat.Port(ports.RestPort + "/tcp")
+	natPlaintextPort := nat.Port(ports.PlaintextPort + "/tcp")
+	config := &container.Config{
+		Image:    imageName,
+		Hostname: "broker",
+		Cmd:      strslice.StrSlice{"bash", "-c", "'/etc/confluent/docker/run'"},
+		ExposedPorts: nat.PortSet{
+			natRestPort:      struct{}{},
+			natPlaintextPort: struct{}{},
+		},
+		Env: getContainerEnvironmentWithPorts(ports),
+	}
+	hostConfig := &container.HostConfig{
+		PortBindings: nat.PortMap{
+			natRestPort: []nat.PortBinding{
+				{
+					HostIP:   localhost,
+					HostPort: ports.RestPort,
+				},
+			},
+			natPlaintextPort: []nat.PortBinding{
+				{
+					HostIP:   localhost,
+					HostPort: ports.PlaintextPort,
+				},
+			},
+		},
+	}
+
+	createResp, err := dockerClient.ContainerCreate(context.Background(), config, hostConfig, nil, platform, "confluent-local")
+	if err != nil {
+		return errors.CatchContainerNameInUseError(err)
+	}
+	log.CliLogger.Tracef("Create confluent-local container success")
+
+	err = dockerClient.ContainerStart(context.Background(), createResp.ID, types.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+
+	output.Printf("Started Confluent local container %v.\nContinue your experience with Confluent local running `confluent local kafka produce` and `confluent local kafka consume`.\n", createResp.ID[:containerIdShortLength])
+	return nil
+}
+
+func (c *command) prepareValidPorts() error {
+	if c.Config.LocalPorts != nil {
+		return nil
+	}
+
+	restPortInt, err := freeport.GetFreePort()
```

Comment:
Ports are always integers, so this is probably a bit redundant. Also "rest" means Kafka REST, right?
```suggestion
	kafkaRestPort, err := freeport.GetFreePort()
```